src/ui/public/kuery/lib/ast.js

```diff
@@ -0,0 +1,58 @@
+import grammar from 'raw!./kuery.peg';
+import PEG from 'pegjs';
+import _ from 'lodash';
+
+const kueryParser = PEG.buildParser(grammar);
+
+export function fromKueryExpression(expression) {
+  if (_.isUndefined(expression)) {
+    throw new Error('expression must be a string, got undefined instead');
+  }
+
+  return kueryParser.parse(expression);
+}
+
+export function toKueryExpression(ast) {
+  if (!Array.isArray(ast)) throw new Error('AST must be an array');
+
+  return ast.reduce((acc, node) => {
+    return `${acc} ${stringifyNode(node)}`;
+  }, '').trim();
+}
+
+function stringifyNode(node) {
+  if (!node || !node.type) return '';
+
+  let retVal = '';
+  if (node.negate) retVal += '-';
+
+  switch (node.type) {
+    case 'match':
+      retVal += `${node.params.field}:${node.params.value}`;
+      break;
+  }
+
+  return retVal;
+}
+
+function addNode(ast, node) {
+  return ast.concat(node);
+}
+
+function buildMatchNode({ field, values, operation }) {
```

Comment:
It does use the same signature so I can easily pass it into those place where the filterManager was used, but it has utility beyond the old filter code. Any code that wants to add a match query to the AST would use this function to build it.

I'll pull it out into a separate file, I just left it in here for convenience since I only had one type so far.